src/vcs/git.rs

```diff
@@ -192,6 +204,65 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    diff.add_created_file(path);
+                },
+                Delta::Deleted => {
+                    let diff_file = delta.old_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    diff.add_deleted_file(path);
+                },
+                Delta::Modified => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    let patch = Patch::from_diff(&git_diff, idx)?;
+
+                    if let Some(patch) = patch {
+                        let mut hunks: Vec<Hunk> = Vec::new();
+
+                        for h in 0..patch.num_hunks() {
+                            let (hunk, hunk_lines) = patch.hunk(h)?;
+                            let header = hunk.header().to_owned();
+                            let mut lines: Vec<LineDiff> = Vec::new();
+
+                            for l in 0..hunk_lines {
+                                let line = patch.line_in_hunk(h, l)?;
+                                lines.push(line.into());
+                            }
+                            hunks.push(Hunk { header, lines });
+                        }
+                        diff.add_modified_file(path, hunks);
+                    } else if diff_file.is_binary() {
+                        // TODO
```

Comment:
How should we handle this?

Reply:
Should we have a `Diff::Binary` and not present those diffs? Just have two blobs?

Reply:
Yeah, changed `FileDiff` to an enum. However for now I'm not including the blobs in the Binary, since they'd have to be owned. I'm a little worried that this runs on a large file and all that data has to be transfered even though we aren't displaying it. I think we should handle that case separately and think more about the use-case first.

Reply:
c5cc0a2